### PR TITLE
Validate string before entering loop

### PR DESCRIPTION
**Pull request by ricjon**

### CODE REVIEW

File: blekingesaldo/src/main/java/se/fourspaces/blekingebalance/HexUtils.java [hunk 1]
```diff
@@ -26,14 +26,19 @@
     public static byte[] hexStringToByteArray(String s) {
         int len = s.length();
         byte[] data = new byte[len / 2];
-        try {
-            for (int i = 0; i < len; i += 2) {
-                data[i / 2] = (byte) ((Character.digit(s.charAt(i), 16) << 4)
-                        + Character.digit(s.charAt(i + 1), 16));
+        String hexStringPattern = "^([0-9A-Fa-f]{2})+$";
```
Comment (eliasson) [inline blekingesaldo/src/main/java/se/fourspaces/blekingebalance/HexUtils.java]: This should be made a constant.

File: blekingesaldo/src/main/java/se/fourspaces/blekingebalance/HexUtils.java [hunk 2]
```diff
@@ -26,14 +26,19 @@
...
+        boolean isHexString = s.matches(hexStringPattern);
```
Comment (eliasson) [inline blekingesaldo/src/main/java/se/fourspaces/blekingebalance/HexUtils.java]: Please use `final` where possible.

**Review by eliasson — CHANGES_REQUESTED** [review]
Thanks for the PR, please see my two comments on code changes and supply a description for the PR in general.

File: blekingesaldo/src/main/java/se/fourspaces/blekingebalance/HexUtils.java (hunk 2 above)
Comment (ricjon) [reply blekingesaldo/src/main/java/se/fourspaces/blekingebalance/HexUtils.java]: Fixed it.

**Review by eliasson — APPROVED** [review]

**Pull request merged**